parlai/core/torch_generator_agent.py

```diff
@@ -624,7 +647,12 @@ def beam_search(self, model, batch, beam_size, start=1, end=2,
             score = F.log_softmax(score, dim=-1)
             for i, b in enumerate(beams):
                 if not b.done():
-                    b.advance(score[i])
+                    partial_hyps = []
+                    if num_iterations > 1:
+                        for j in range(i):
```

Comment:
`for j in (index into beams)` seems strange. Is this intentional?

Reply:
so yes we iterate over all iterations done BEFORE i